core/pom.xml

```diff
@@ -17,8 +18,22 @@
         <dependency>
             <groupId>com.github.docker-java</groupId>
             <artifactId>docker-java</artifactId>
-            <version>2.2.0</version>
+            <version>3.0.0</version>
+            <exclusions>
+                <!-- replace with junixsocket -->
+                <exclusion>
+                    <groupId>de.gesellix</groupId>
+                    <artifactId>unix-socket-factory</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>org.rnorth</groupId>
+            <artifactId>tcp-unix-socket-proxy</artifactId>
```

Comment:
Until netty and/or docker-java have native support for unix domain sockets, use a slightly hacky but effective workaround of proxying over TCP. For Testcontainers' purposes, there's no perceptible performance impact.